akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -843,10 +850,10 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
       // Don't mark gossip state as seen while exiting is in progress, e.g.
       // shutting down singleton actors. This delays removal of the member until
       // the exiting tasks have been completed.
-      if (exitingTasksInProgress)
-        latestGossip = winningGossip
-      else
-        latestGossip = winningGossip seen selfUniqueAddress
+      latestGossip =
+        (if (exitingTasksInProgress) winningGossip
+        else winningGossip seen selfUniqueAddress)
+          .pruneTombstones(System.currentTimeMillis() - PruneGossipTombstonesAfter.toMillis)
```

Comment:
Thinking of if this is the right place to do the pruning. Would an alternative be to let the leader do it as part of the periodic leader actions, and when something was pruned it would bump the vector clock to spread that to others. Worst case tombstones would be reintroduced if there is a merge conflict but then leader would remove them again later.